### PR TITLE
Coded index cleanup

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@nguerrera

File: src/System.Reflection.Metadata/tests/Metadata/Ecma335/CodedIndexTests.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            Assert.Equal(0x20 | 18, CodedIndex.HasCustomDebugInformation(MetadataTokens.ManifestResourceHandle(1)));
+            Assert.Equal(0x20 | 19, CodedIndex.HasCustomDebugInformation(MetadataTokens.GenericParameterHandle(1)));
+            Assert.Equal(0x20 | 20, CodedIndex.HasCustomDebugInformation(MetadataTokens.GenericParameterConstraintHandle(1)));
+            Assert.Equal(0x20 | 21, CodedIndex.HasCustomDebugInformation(MetadataTokens.MethodSpecificationHandle(1)));
+            Assert.Equal(0x20 | 22, CodedIndex.HasCustomDebugInformation(MetadataTokens.DocumentHandle(1)));
+            Assert.Equal(0x20 | 23, CodedIndex.HasCustomDebugInformation(MetadataTokens.LocalScopeHandle(1)));
+            Assert.Equal(0x20 | 24, CodedIndex.HasCustomDebugInformation(MetadataTokens.LocalVariableHandle(1)));
+            Assert.Equal(0x20 | 25, CodedIndex.HasCustomDebugInformation(MetadataTokens.LocalConstantHandle(1)));
+            Assert.Equal(0x20 | 26, CodedIndex.HasCustomDebugInformation(MetadataTokens.ImportScopeHandle(1)));
+        }
```
Comment (nguerrera) [inline src/System.Reflection.Metadata/tests/Metadata/Ecma335/CodedIndexTests.cs]: It would probably be worthwhile to have some tests with RID != 1.

**Comment by nguerrera** [comment]
:+1:

File: src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/CodedIndex.cs [hunk 2]
```diff
@@ -6,38 +6,169 @@ namespace System.Reflection.Metadata.Ecma335
...
+        /// <see cref="AssemblyReferenceHandle"/>,
+        /// <see cref="AssemblyFileHandle"/>,
+        /// <see cref="ExportedTypeHandle"/>,
+        /// <see cref="ManifestResourceHandle"/>,
+        /// <see cref="GenericParameterHandle"/>,
+        /// <see cref="GenericParameterConstraintHandle"/> or
+        /// <see cref="MethodSpecificationHandle"/>.
+        /// </param>
+        /// <exception cref="ArgumentException">Unexpected handle kind.</exception>
+        public static int HasCustomAttribute(EntityHandle handle) => (handle.RowId << (int)HasCustomAttributeTag.BitCount) | (int)ToHasCustomAttributeTag(handle.Kind);
```
Comment (tmat) [inline src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/CodedIndex.cs]: Nah, github is just badly designed and shouldn't be used as a measure of reasonable line length, imo. if I open the source file in VS and resize the window to see the same I see here I get very small window. Nobody has such a small monitor.

File: src/System.Reflection.Metadata/tests/Metadata/Ecma335/CodedIndexTests.cs (hunk 1 above)
Comment (tmat) [inline src/System.Reflection.Metadata/tests/Metadata/Ecma335/CodedIndexTests.cs]: OK

**Pull request merged**